caffe2/core/c10_operator.h

```diff
@@ -0,0 +1,96 @@
+#pragma once
+
+#include <vector>
+#include <ATen/core/dispatch/OpSchemaRegistration.h>
+#include <ATen/core/dispatch/KernelRegistration.h>
+#include <ATen/core/function_schema.h>
+
+namespace caffe2 {
+namespace detail {
+
+template<class Caffe2Operator> const c10::OperatorHandle& c10_op_handle_for_c2_op();
+template <class Caffe2Operator>
+void call_caffe2_op_from_c10(c10::Stack* stack, c10::KernelCache* cache) { // TODO Pass in correct cache type
+  // precondition: on the stack, there's an IValue for each input and an IValue for each output.
+  // The output ones could either be a preallocated tensor or ivalue::None.
+
+  const auto& schema = c10_op_handle_for_c2_op<Caffe2Operator>().schema();
+  const size_t num_outputs = schema.returns().size();
+  const size_t total_num_arguments = schema.arguments().size();
+  const size_t num_inputs = total_num_arguments - num_outputs;
+
+  c10::ArrayRef<c10::IValue> inputs = torch::jit::peekSlice(*stack, 0, num_inputs, total_num_arguments);
+  c10::ArrayRef<c10::IValue> outputs = torch::jit::peekSlice(*stack, num_inputs, num_outputs, total_num_arguments);
+
+  auto inputsVec = inputs.vec();
+  auto outputsVec_ = outputs.vec();
+
+  const auto device = at::Device(at::DeviceType::CPU); // TODO Handle GPU devices
+
+  for (auto& output : outputsVec_) {
+    if (output.isNone() || (output.isTensor() && !output.toTensor().defined())) {
+      output = at::Tensor(c10::C10Tensor(caffe2::empty({0}, device)));
```

Comment:
Will this really work in general? You're implicitly allocating a float tensor. What about non-float outputs?

Reply:
Most caffe2 operators will handle this gracefully. But as discussed offline, a better solution would be to make caffe2 `OperatorBase::Output()` work correctly when the tensor is fully uninitialized and then pass in uninitialized tensors here. Will probably not fix it in this PR though.

Reply:
Can we get a comment here to that effect, then?